### PR TITLE
fix: adjust docs page for custom url

### PR DESCRIPTION
**Pull request by caleb-at-pieces**
This pull request aims to help clarify to our users when and how they should use the custom url feature in vscode

1. advise against NGROK (most users actually need this for things like dev containers which are on the same LAN so they really shouldn't use NGROK)
2. advise against using the feature. I have multiple sentry errors from people who think that this is their pieces cloud url... etc. I don't know if we need to rename the setting, but in general I am trying to make them really think about if they need to use this feature before they do.
3. we also need to ensure we are showing people the correct port(s) according to their OS

preview: https://docs.pieces.app/~257

### CODE REVIEW

File: docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx [hunk 1]
```diff
@@ -11,6 +11,8 @@ This guide will walk you through how to connect to a remote Pieces OS instance u
 
 > - This is currently only available for the Pieces for Developers VS Code extension. Functionality for the rest of the Pieces for Developers ecosystem is coming soon.
 
+<Info>If Pieces OS is running on the same machine as VS Code, there should be no requirement to use this feature. Please verify these steps are required for your set up before following this guide. This custom URL is not the same thing as your Pieces Cloud custom url.</Info>
```
Comment (mason-at-pieces) [inline docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx:14]: I would probably put this right under the "Connecting to a Remote Pieces OS Instance" header so its one of the first things people read. Thoughts?

File: docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx [hunk 2]
```diff
@@ -27,25 +29,33 @@ This guide will walk you through how to connect to a remote Pieces OS instance u
...
+  <TabItem value="lan-ip-address">
+    1. Ensure that there is not a firewall in place that will block connection between two machines on your LAN.
+    2. Locate your machine's LAN IP address
+      a. If possible, you should also set up a DHCP IP reservation for your machine running Pieces OS so this IP address will not change in the future.
+    3. For linux your custom url will be `http://{'{LAN IP ADDRESS}'}:5323` and on MacOS or windows it will be `http://{'{LAN IP ADDRESS}'}:1000`
+  </TabItem>
   <TabItem value="static-url">
     1. Install ngrok on the machine with Pieces OS if it isn’t already
-    2. Run the following command on your machine’s terminal: `ngrok http --domain=STATIC_URL_HERE 1000`
+    2. Run the following command on your machine’s terminal: (windows and macos)`ngrok http --domain=STATIC_URL_HERE 1000` (linux)`ngrok http --domain=STATIC_URL_HERE 5323`
```
Comment (mason-at-pieces) [inline docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx:53]: Can we put the (windows/macos) and (linux) on separate lines? It's a bit hard to actually see

File: docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx [hunk 3]
```diff
@@ -27,25 +29,33 @@ This guide will walk you through how to connect to a remote Pieces OS instance u
...
     3. Copy the forwarding URL it generates to use in the next step
   </TabItem>
   <TabItem value={"generated-url"}>
     1. Install ngrok on the machine with Pieces OS if it isn’t already
-    2. Run the following command on your machine’s terminal: `ngrok http 1000`
+    2. Run the following command on your machine’s terminal: (windows and macos)`ngrok http 1000` (linux)`ngrok http 5323`
```
Comment (mason-at-pieces) [inline docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx:58]: Same here. Can we put the (windows/macos) and (linux) on separate lines? It's a bit hard to actually see

**Review by mason-at-pieces — CHANGES_REQUESTED** [review]

**Comment by caleb-at-pieces** [comment]
@mason-at-pieces i also have layered in some functionality to make dev containers work out of the box, not sure where we should put it.

Generally if they have issues with dev containers post the next release it's one of two things:

1. they need to make sure that Pieces os is running *before* they launch the extension, launch PIeces os and reload the vscode window.
2. they need to make sure that there is no firewall blocking requests between machines on LAN

File: docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx (hunk 1 above)
Comment (caleb-at-pieces) [reply docs/product-highlights-and-benefits/connecting-to-a-remote-pieces-os-instance.mdx:14]: I have a habit of skipping at least 1/3 of the webpage cause most of the time it's just garbage info up at the top but I'm down

**Comment by caleb-at-pieces** [comment]
> It seems like we could solve this with some copy and structure updates to the page. It was originally made for the ngrok flow but now it seems like we have many different flows, and now most users who might think they need this won't even need to do any additional setup.
> 
> What are your thoughts @caleb-at-pieces on what you want a user to get from this page?

I agree to a certain extent & you are correct in saying this:

1. this page will be seen by anyone who is using vscode in a remote environment & has connection issues
2. probably first we want to talk about how wsl and dev containers should work completely normally, and if they have issues they should try and set up their inbound firewall rules. it seems like the guide we have for wsl is solving problems for folks
3. last on the page, we push them to use the custom url feature however it should not immediately funnel people into using NGROK as this is not only insecure but it's also not required for a lot of folks as they still are connecting into something that is on their LAN

**Comment by mason-at-pieces** [comment]
This is currently on hold due to onboarding updates and waiting on the video from caleb

**Comment by caleb-at-pieces** [comment]
> @caleb-at-pieces @mason-at-pieces are we still pending on a video here?

yes it should be done this week

**Comment by shivay-at-pieces** [comment]
https://www.youtube.com/watch?v=sbgIaK9kpu4 

@jordan-pieces @caleb-at-pieces

**Comment by mason-at-pieces** [comment]
@shivay-at-pieces @caleb-at-pieces What's the status of this? The videos should be uploaded to YT

**Comment by caleb-at-pieces** [comment]
I will be taking a look today and we can then merge this!

**Comment by caleb-at-pieces** [comment]
i'm 🟢

**Review by mason-at-pieces — APPROVED** [review]
looking good

**Pull request merged**